napari/__main__.py

```diff
@@ -187,5 +188,46 @@ def main():
             )
 
 
+def _run_pythonw():
+    """Execute this script again through pythonw.
+    This ensures we're using a framework build of Python on macOS.
+    """
+    import pathlib
+    import subprocess
+
+    cwd = pathlib.Path.cwd()
+    python_path = pathlib.Path(sys.exec_prefix) / 'bin' / 'pythonw'
+
+    if not python_path.exists():
+        msg = (
+            'pythonw executable not found. '
+            'Please install python.app via conda.'
+        )
+        raise RuntimeError(msg)
+
+    cmd = [python_path, '-m', 'napari']
+
+    # Append command line arguments.
+    if len(sys.argv) > 1:
+        cmd.append(*sys.argv[1:])
+
+    env = os.environ.copy()
+    env["NAPARI_RUNNING_PYTHONW"] = "True"
+
+    subprocess.run(cmd, env=env, cwd=cwd)
+    sys.exit()
```

Comment:
does this exit hide any exit code? the subprocess is run from the previous line, should we inherit the exit code from there?

Reply:
good point.  I think you can do:
```python
    result = subprocess.run(cmd, env=env, cwd=cwd)
    sys.exit(result.returncode)
```
https://docs.python.org/3/library/subprocess.html#subprocess.CompletedProcess